src/KubernetesClient/Informers/SharedOptionsInformer.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using k8s.Informers.Notifications;
+
+namespace k8s.Informers
+{
+    /// <summary>
+    ///     Manages multiple <see cref="SharedInformer{TKey,TResource}" /> for each unique set of <typeparamref name="TOptions" /> and ensures subscriptions are attached to correct one
+    /// </summary>
+    /// <typeparam name="TResource"></typeparam>
+    /// <typeparam name="TOptions"></typeparam>
+    public class SharedOptionsInformer<TResource, TOptions> : IInformer<TResource, TOptions>
```

Comment:
I presume this type is supposed to be thread-safe? It's assumed to be so in the informer example as (the derived) `SharedKubernetesInformer<,>` is registered as a singleton.